demo/builder/main.go

```diff
@@ -0,0 +1,167 @@
+package main
```

Comment:
Can we build the demo into CI (Actions) so we know it works and does not break over time?

Reply:
This demo is a server-side example, which is somewhat similar to mock. I personally think that it is not meaningful to test alone, but it is too difficult for me to test with plug-ins.

Reply:
Yup but we can at least try to compile, run and query the server in CI.

Since I don't know Go and will have to maintain this code after it gets merged, I'd like to have a guarantee it does not break.